frontend/app/utils/fetch-utils.ts

```diff
@@ -24,11 +25,14 @@ export interface FetchFunctionInit extends RequestInit {
  * Return a custom fetch() function if a proxy URL has been provided.
  * If no proxy has been provided, simply return global.fetch().
  */
-export function getFetchFn(proxyUrl?: string) {
+export function getFetchFn(proxyUrl?: string, timeout?: number) {
   if (proxyUrl) {
-    log.debug('A proxy has been configured: [%s]; using custom fetch', proxyUrl);
+    if (timeout === undefined) {
+      ({ HTTP_PROXY_TLS_TIMEOUT: timeout } = getEnv());
+    }
+    log.debug('A proxy has been configured with timeout: [%s], [%d] milliseconds; using custom fetch', proxyUrl, timeout);
```

Comment:
Method argument `timeout` should be immutable.
```suggestion
    const { HTTP_PROXY_TLS_TIMEOUT } = getEnv()
    const proxyTlsTimeout = timeout ?? HTTP_PROXY_TLS_TIMEOUT;
    log.debug('A proxy has been configured with timeout: [%s], [%d] milliseconds; using custom fetch', proxyUrl, proxyTlsTimeout);
```
```suggestion
    if (timeout === undefined) {
      ({ HTTP_PROXY_TLS_TIMEOUT: timeout } = getEnv());
    }
    log.debug('A proxy has been configured with timeout: [%s], [%d] milliseconds; using custom fetch', proxyUrl, timeout);
```